storage/innobase/handler/ha_innodb.cc

```diff
@@ -2193,6 +2193,9 @@ convert_error_code_to_mysql(
 		return(HA_ERR_RECORD_FILE_FULL);
 
 	case DB_TEMP_FILE_WRITE_FAIL:
+		innodb_transaction_abort(thd,
+					 innobase_rollback_on_timeout,
+					 error);
```

Comment:
Is this change really necessary? It is not present in #3903. I suggested this when we were discussing this. As far as I understand, this error could be flagged only during the execution of `copy_data_between_tables()` in `ALTER TABLE…ALGORITHM=COPY` or similar statements, but not during DML. For DML, we realized that if a write to a temporary file fails in trx_t::apply_log()`, then we should flag an error to the `ha_innobase::commit_inplace_alter_table()` thread and let the DML transaction commit normally.

Reply:
This DB_TEMP_FILE_WRITE_FAIL error could happen during COPY  algorithm or bulk insert. This code addresses
the issue only for these scenario. For online alter, concurrent DML just marks the index as corrupted and move on.